Running Auroraboot with no params gives a kernel panic

```
$ go run main.go                                      
2024-11-19T10:30:44+01:00 INF Downloading https://github.com//releases/download//core--.iso...
2024-11-19T10:30:44+01:00 INF Downloading https://github.com//releases/download//core--.squashfs...
2024-11-19T10:30:44+01:00 INF Downloading https://github.com//releases/download//core---initrd...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe8601f]

goroutine 31 [running]:
github.com/kairos-io/AuroraBoot/pkg/ops.download({0x14fd000, 0x1e34fc0}, {0xc0004b7a00, 0x36}, {0xc0004993c0, 0x1c})
	/home/itxaka/projects/AuroraBoot/pkg/ops/network.go:57 +0x21f
github.com/kairos-io/AuroraBoot/deployer.(*Deployer).StepDownloadSquashFS.DownloadArtifact.func4({0x14fd000?, 0x1e34fc0?})
	/home/itxaka/projects/AuroraBoot/pkg/ops/network.go:42 +0x2d
github.com/spectrocloud-labs/herd.(*Graph).Run.func3({0x14fd000?, 0x1e34fc0?}, 0xc0005d1470, {0x117acdd, 0x11}, 0x0?)
	/home/itxaka/go/pkg/mod/github.com/spectrocloud-labs/herd@v0.4.2/dag.go:159 +0x43
created by github.com/spectrocloud-labs/herd.(*Graph).Run in goroutine 1
	/home/itxaka/go/pkg/mod/github.com/spectrocloud-labs/herd@v0.4.2/dag.go:158 +0x633
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe8601f]

goroutine 29 [running]:
github.com/kairos-io/AuroraBoot/pkg/ops.download({0x14fd000, 0x1e34fc0}, {0xc0004b7980, 0x34}, {0xc000499380, 0x1a})
	/home/itxaka/projects/AuroraBoot/pkg/ops/network.go:57 +0x21f
github.com/kairos-io/AuroraBoot/deployer.(*Deployer).StepDownloadInitrd.DownloadArtifact.func3({0x14fd000?, 0x1e34fc0?})
	/home/itxaka/projects/AuroraBoot/pkg/ops/network.go:42 +0x2d
github.com/spectrocloud-labs/herd.(*Graph).Run.func3({0x14fd000?, 0x1e34fc0?}, 0xc0005d1470, {0x1177e54, 0xf}, 0x0?)
	/home/itxaka/go/pkg/mod/github.com/spectrocloud-labs/herd@v0.4.2/dag.go:159 +0x43
created by github.com/spectrocloud-labs/herd.(*Graph).Run in goroutine 1
	/home/itxaka/go/pkg/mod/github.com/spectrocloud-labs/herd@v0.4.2/dag.go:158 +0x633
exit status 2
```

There should be a safety check that if we are not passing a release, an iso, or a container, we dont start aurora and exit gracefully. Instead this is trying to download the release isos with no versions and panicking

--- a/internal/cmd/app.go
+++ b/internal/cmd/app.go
@@ -57,6 +57,9 @@ func GetApp(version string) *cli.App {
 			if err != nil {
 				return err
 			}
+			if r.ContainerImage == "" && (r.Repository == "" || r.ReleaseVersion == "" || r.ArtifactVersion == "" || r.Flavor == "") {
+				return errors.New("no source defined: provide container_image or a complete release artifact configuration")
+			}
 
 			c.ISO.HandleDeprecations(internal.Log)
 

--- a/internal/cmd/app_test.go
+++ b/internal/cmd/app_test.go
@@ -1,0 +1,29 @@
+package cmd_test
+
+import (
+	"bytes"
+
+	cmdpkg "github.com/kairos-io/AuroraBoot/internal/cmd"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("root command", Label("cmd"), func() {
+	It("rejects an invocation without an image or release source", func() {
+		app := cmdpkg.GetApp("v0.0.0")
+		app.Writer = new(bytes.Buffer)
+
+		err := app.Run([]string{"auroraboot"})
+
+		Expect(err).To(MatchError("no source defined: provide container_image or a complete release artifact configuration"))
+	})
+
+	It("rejects an incomplete release source", func() {
+		app := cmdpkg.GetApp("v0.0.0")
+		app.Writer = new(bytes.Buffer)
+
+		err := app.Run([]string{"auroraboot", "--set", "repository=kairos-io/kairos"})
+
+		Expect(err).To(MatchError("no source defined: provide container_image or a complete release artifact configuration"))
+	})
+})

